comment-updater-headless/src/main/kotlin/PluginRunner.kt

```diff
@@ -0,0 +1,281 @@
+import com.intellij.ide.impl.ProjectUtil
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.application.ApplicationStarter
+import com.intellij.openapi.components.ServiceManager
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.vcs.ProjectLevelVcsManager
+import com.intellij.openapi.vcs.VcsException
+import com.intellij.openapi.vcs.changes.Change
+import com.intellij.openapi.vcs.impl.ProjectLevelVcsManagerImpl
+import git4idea.GitCommit
+import git4idea.GitVcs
+import git4idea.repo.GitRepositoryManager
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.runBlocking
+import org.jetbrains.research.commentupdater.dataset.DatasetExample
+import org.jetbrains.research.commentupdater.models.MetricsCalculator
+import org.jetbrains.research.commentupdater.processors.ProjectMethodExtractor
+import org.jetbrains.research.commentupdater.processors.RefactoringExtractor
+import org.jetbrains.research.commentupdater.utils.qualifiedName
+import org.jetbrains.research.commentupdater.utils.textWithoutDoc
+import java.io.File
+import java.util.concurrent.CountDownLatch
+import kotlin.concurrent.thread
+import kotlin.system.exitProcess
+
+
+class PluginRunner: ApplicationStarter {
+    override fun getCommandName(): String = "CommentUpdater"
+    override fun main(args: Array<String>) {
+        CodeCommentExtractor().main(args)
+    }
+}
+
+
+class CodeCommentExtractor {
+
+    // Reading project paths
+    private val inputPath = HeadlessConfig.INPUT_FILE
+
+    private val exampleWriter = ExampleWriter()
+
+    private val statsHandler = StatisticHandler()
+
+    // Metric model
+    private val metricsModel = MetricsCalculator()
+
+
+    companion object {
+        enum class LogLevel {
+            INFO, WARN, ERROR
+        }
+
+        var projectTag: String = ""
+        var projectProcess: String = ""
+
+        fun  log(level: LogLevel, message: String, logThread: Boolean = false,
+                 applicationTag: String = "[HeadlessCommentUpdater]") {
+            val fullLogMessage = "$level ${if (logThread) Thread.currentThread().name else ""} $applicationTag [$projectTag $projectProcess] $message"
+
+            when (level) {
+                LogLevel.INFO -> {
+                    println(fullLogMessage)
```

Comment:
I thought we use a logger from the IntelliJ platform...

Reply:
As I understood, IntelliJ plugin automatically copies std output to the idea.log file. So, we don't have to duplicate logs in file and in std output.